include/nbl/builtin/hlsl/tonemapper/operators.hlsl

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_TONE_MAPPER_OPERATORS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_TONE_MAPPER_OPERATORS_INCLUDED_
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace tonemapper
+{
+
+struct ReinhardParams
+{
+	using this_t = ReinhardParams;
+	static this_t create(float EV, float key = 0.18f, float WhitePointRelToEV = 16.f)
+	{
+		this_t retval;
+		retval.keyAndManualLinearExposure = key * exp2(EV);
+		retval.rcpWhite2 = 1.f / (WhitePointRelToEV * WhitePointRelToEV);
+		return retval;
+	}
+
+	float32_t keyAndManualLinearExposure;
+	float32_t rcpWhite2;
+};
+
+struct ACESParams
+{
+	using this_t = ACESParams;
+	static this_t create(float EV, float key = 0.18f, float Contrast = 1.f) {
+		this_t retval;
+		retval.gamma = Contrast;
+		retval.exposure = EV + log2(key * 0.77321666f);
+		return retval;
+	}
+
+	float32_t gamma; // 1.0
+	float32_t exposure; // actualExposure+midGrayLog2
+};
+
+
+float32_t3 reinhard(ReinhardParams params, float32_t3 rawCIEXYZcolor)
+{
+	float32_t exposureFactors = params.keyAndManualLinearExposure;
+	float32_t exposedLuma = rawCIEXYZcolor.y * exposureFactors;
+	float32_t colorMultiplier = (exposureFactors * (1.0 + exposedLuma * params.rcpWhite2) / (1.0 + exposedLuma));
+	return rawCIEXYZcolor * colorMultiplier;
+}
+
+float32_t3 aces(ACESParams params, float32_t3 rawCIEXYZcolor)
+{
```

Comment:
these should be `operator()(const float32_t3 rawCIEXYZcolor)` inside the respective Reinhard and ACES structs

Reply:
Done